unity/Assets/Scripts/AgentManager.cs

```diff
@@ -766,6 +766,44 @@ private void Update()
             payload.Add(new KeyValuePair<string, byte[]>("image", captureScreen()));
 		}
 	}
+	
+	private void resetImageSynthesis(Camera camera)
+	{
+		ImageSynthesis imageSynthesis = camera.gameObject.GetComponentInChildren<ImageSynthesis>();
+		if (imageSynthesis != null && imageSynthesis.enabled)
+		{
+			imageSynthesis.OnCameraChange();
+			imageSynthesis.OnSceneChange();
+		}
+	}
+
+	private void resetAllImageSynthesis()
+	{
+		foreach (var agent in this.agents)
+		{
+			resetImageSynthesis(agent.m_Camera);
+		}
+
+		foreach (var camera in this.thirdPartyCameras)
+		{
+			resetImageSynthesis(camera);
+		}
+	}
+	
+	public IEnumerator WaitOnResolutionChange(int width, int height) {
+		while (Screen.width != width || Screen.height != height) {
+			yield return null;
+		}
+		this.resetAllImageSynthesis();
```

Comment:
Is it possible that `resetAllImageSynthesis` causes a change in which colors represent which objects for the instance/semantic segmentation frames? I remember the code for doing this assignment was a bit complex.

Reply:
It isn't possible (at least in this case) for the colors to change since we take the MD5 of the ObjectId of the SimObject, which can't change during the ChangeResolution action.  Just to protect us a bit in the future I added assertions in test_unity.py that compare the pre and post color_to_object_id dictionaries on the event.

Reply:
👍  I suppose I was more worried that someone would call an action which reassigns all object ids without reassigning the colors (does this ever happen?), grabs the mapping of colors->objects, and then `ChangeResolution` (which would then update all the colors). I suppose this isn't really an issue with this PR but it would be nice if we had a method for assigning colors that was more invariant to these types of issues (as the object id seems to be more of an "object id" nowadays).

Reply:
I'm not sure if thats possible since the the ObjectId is assigned when the scene loads based off the position and even if the object moves, the objectId remains the same.  But your point is valid, since even the slightest change in the scene to the object results in a new ID and a new color.